src/rust/bitbox02/src/backup.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright 2020 Shift Crypto AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+extern crate alloc;
+use alloc::string::String;
+pub use bitbox02_sys::backup_error_t as Error;
+
+pub struct CheckData {
+    pub id: String,
+    pub name: String,
+    pub birthdate: u32,
+}
+
+/// Returns the backup id, name and birthdate of the backup that matches the current keystore seed.
+/// If none matches, `Err()` is returned.
+pub fn check() -> Result<CheckData, Error> {
+    let mut id = [0u8; 65];
```

Comment:
`backup_check` in https://github.com/digitalbitbox/bitbox02-firmware/blob/75470b6/src/backup/backup.h#L34 says something odd:

> id_out must have max 256 bytes in size; hww.options BackupInfo.id

but looking at backup.c, it clearly is 32 bytes in hex. I guess the comment in `backup_check` in backup.h is old and needs to be updated.

Reply:
ah, never mind. I see it's updated in the last commit 51bb552.